src/Sulu/Bundle/ReferenceBundle/UserInterface/Controller/Admin/ReferenceController.php

```diff
@@ -85,14 +85,6 @@ public function cgetAction(Request $request): Response
             'offset' => $offset,
         ]);
 
-        /** @var string|null $sortBy */
-        $sortBy = $request->query->get('sortBy');
-        /** @var 'asc'|'desc' $sortOrder */
-        $sortOrder = $request->query->get('sortOrder', 'asc');
-        $sortBys = $sortBy ? [
-            $sortBy => $sortOrder,
-        ] : [];
-
         $fields = \explode(',', $request->query->get('fields', ''));
         $removeFields = ['id']; // the frontend always add the id field, but we don't need it in this case as we group by other fields
         $fields = [
```

Comment:
We should also validate the `fields` with the list metadata, otherwise in it is still possible to crash, when the FE sends invalid fields.
Both `fields` and `sortBy` are directly used from the request and unfortunately it is also not validated in the FE 🙈

Reply:
Yes, I don't know why we can't use some parts of the `DoctrineListBuilder` here as that should have all the necessary information.

Reply:
@mamazu I asked the same thing @alexander-schranz 😂 

I don’t recall the details exactly, but I believe we were considering an alternative approach at that time. This approach would have utilized the ⁠DoctrineListBuilder in the repository. Unfortunately, time slipped away, and we didn’t have the opportunity to thoroughly discuss or refactor it before the release 🙈.

I’m still open to refactoring it to use the DoctrineListBuilder, so we can maintain some consistency between the controllers.